drivers/hmis/app/models/hmis/ce/match/internal/client_pool_evaluator.rb

```
@@ -60,10 +65,18 @@ def call(client)
     # evaluate!('current_age >= 65 AND veteran_status = 1', {current_age: 50, veteran_status: 1})
     def eval_requirement(client_values)
       @calculator.evaluate!(@pool.requirement_expression, **client_values)
+    rescue Dentaku::Error, ArgumentError => e
```

I was seeing some errors with evaluating clients that had nil values, such as nil DOB/age. I _think_ they are arising now because usually those clients would've been prefiltered out. In this case, we are just evaluating against one client at a time, so prefiltering doesn't make sense, but that means the evaluator needs to become more flexible to values that can't be compared.

Let me know if you have any concerns about the behavior I've introduced here!

---

I think it would be better not to swallow exceptions if we can avoid it. Maybe it's okay for now but I think we should correct the calculator to handle nils if possible (same issue in `eval_priority`)

---

Remove this change from this PR, make a ticket

---

Papertrail https://github.com/open-path/Green-River/issues/8195